tools/metamorpheus/macros.xml

```diff
@@ -0,0 +1,712 @@
+                <param argument="TotalPartitions" type="integer" value="1" label="TotalPartitions"/>
+                <param argument="PrecursorMassTolerance" type="float" value="15.0000" label="PrecursorMassTolerance in PPM"/>
+                <param argument="ProductMassTolerance" type="float" value="25.0000" label="ProductMassTolerance in PPM"/>
+                <param argument="AddCompIons" type="boolean" truevalue="true" falsevalue="false" checked="false" label="AddCompIons"/>
+                <param argument="ScoreCutoff" type="float" value="5.0" label="ScoreCutoff"/>
+                <param argument="ReportAllAmbiguity" type="boolean" truevalue="true" falsevalue="false" checked="true" label="ReportAllAmbiguity"/>
+                <param argument="NumberOfPeaksToKeepPerWindow" type="integer" value="200" label="NumberOfPeaksToKeepPerWindow"/>
+                <param argument="MinimumAllowedIntensityRatioToBasePeak" type="float" value="0.01" label="MinimumAllowedIntensityRatioToBasePeak"/>
+                <param argument="NormalizePeaksAccrossAllWindows" type="boolean" truevalue="true" falsevalue="false" checked="false" label="NormalizePeaksAccrossAllWindows"/>
+                <param argument="TrimMs1Peaks" type="boolean" truevalue="true" falsevalue="false" checked="false" label="TrimMs1Peaks"/>
+                <param argument="TrimMsMsPeaks" type="boolean" truevalue="true" falsevalue="false" checked="true" label="TrimMsMsPeaks"/>
+                <param argument="UseDeltaScore" type="boolean" truevalue="true" falsevalue="false" checked="false" label="UseDeltaScore"/>
+                <param argument="QValueOutputFilter" type="float" value="1.0" label="QValueOutputFilter"/>
+                <param argument="MaxHeterozygousVariants" type="integer" value="4" label="MaxHeterozygousVariants"/>
+                <param argument="MinVariantDepth" type="integer" value="1" label="MinVariantDepth"/>
+                <param argument="SeparationType" type="select" label="SeparationType">
+                    <option value="HPLC" selected="true">HPLC</option>
+                    <option value="CZE">CZE</option>
+                </param>
+                <param argument="DissociationType" type="select" label="DissociationType">
+                    <expand macro="dissociation_options"/>
+                </param>
+                <param argument="MS2ChildScanDissociationType" type="select" label="MS2ChildScanDissociationType">
+                    <expand macro="dissociation_options"/>
+                </param>
+                <param argument="MS3ChildScanDissociationType" type="select" label="MS2ChildScanDissociationType">
+                    <expand macro="dissociation_options"/>
+                </param>
+            </when>
+        </conditional>
+    </xml>
+    <token name="@COMMON_CONFIG@">
+[CommonParameters]
+TaskDescriptor = "CalibrateTask"
+MaxThreadsToUsePerFile = 11
+#if $common.set_common == 'yes' 
```

Comment:
I would prefer to use sections instead of conditionals. Then you may skip this if-else and just use the defaults in the tool form (I guess they are the same anyway).

The disadvantage of conditionals is that they can not be changed in workflows.

Also we would know if there are no typos in the code that is currently in the if branch and the sytnax is correct (ie accepted by MM).

Reply:
These config yaml files may change by the version. They can be generated with `metamorpheus -g`, so if possible, it might be worth adding a test to make sure the default parameters and parameter names haven't changed since the last version.